### PR TITLE
feat(pdk): add `kong.client.tls.set_client_ca_list`

### PR DESCRIPTION
**Pull request by catbro666**
Add a function binding to the API `set_client_ca_list` of lua-kong-nginx-module module. Simple phase check test is added. The API itself was already tested inside lua-kong-nginx-module.

This PR relies on [lua-kong-nginx-module #46](https://github.com/Kong/lua-kong-nginx-module/pull/46) (which is already released in 0.3.0), and is relied by [kong-ee #3890](https://github.com/Kong/kong-ee/pull/3890)

FTI-4430

### CODE REVIEW

**Comment by kikito** [comment]
Hi, could you rebase this over master and add a test for this function in the `/t` folder ?

**Comment by catbro666** [comment]
> 

Rebased. The API itself was already tested inside lua-kong-nginx-module and the PDK
function is simply a wrapper around it, so only phase check test is needed to add.

File: kong/pdk/client/tls.lua [hunk 1]
```diff
@@ -85,6 +85,43 @@ local function new()
...
+  -- local x509, err = x509_lib.new(pem_cert, "PEM")
+  -- -- err check
+  -- res, err = chain:add(x509)
+  -- -- err check
+  -- -- `chain.ctx` is the raw data of the chain, i.e. `STACK_OF(X509) *`
+  -- res, err = kong.client.tls.set_client_ca_list(chain.ctx)
+  -- if not res then
+  --   -- do something with err
+  -- end
+  function _TLS.set_client_ca_list(ca_list)
```
Comment (mayocream) [inline kong/pdk/client/tls.lua:118]: Shall we add a unit test for this function? Either in `t/` or `spec/.`

Comment (catbro666) [reply kong/pdk/client/tls.lua:118]: @mayocream Tests added.

**Review by chronolaw — APPROVED** [review]

File: kong/pdk/client/tls.lua [hunk 2]
```diff
@@ -85,6 +85,43 @@ local function new()
...
+  --
+  -- The downstream client then can use this DN information to filter certificates,
+  -- and chooses an appropriate certificate issued by a CA in the list.
+  --
+  -- The type of `ca_list` paramter is `STACK_OF(X509) *` which can be created by
+  -- using the API of `resty.openssl.x509.chain` or `parse_pem_cert()` of `ngx.ssl`
+  --
+  -- @function kong.client.tls.set_client_ca_list
+  -- @phases certificate
+  -- @treturn true|nil Returns `true` if successful, `nil` if it fails.
```
Comment (fffonion) [inline kong/pdk/client/tls.lua:100]: missing @tparam

Comment (catbro666) [reply kong/pdk/client/tls.lua:101]: added

**Pull request merged**